src/java/org/apache/cassandra/io/sstable/SSTable.java

```diff
@@ -81,6 +80,35 @@
     public static final int TOMBSTONE_HISTOGRAM_SPOOL_SIZE = 100000;
     public static final int TOMBSTONE_HISTOGRAM_TTL_ROUND_SECONDS = Integer.valueOf(System.getProperty("cassandra.streaminghistogram.roundseconds", "60"));
 
+    /**
+     * Watcher used when opening sstables to discover extra components, eg. archive component
+     */
+    public static volatile SSTableWatcher sstableWatcher = new SSTableWatcher() {};
```

Comment:
I don't think we can use public static volatile field for this - cannot it be obtained from storage handler or storage provider ?

Reply:
they are the `metadata` on top of remote file system, like `LogTransaction`. should not be part of storage provider. 

I have moved it to a separate interface with config `cassandra.custom_sstable_watcher`